packages/lab/src/list-next/useList.ts

```diff
@@ -0,0 +1,287 @@
+import {
+  Children,
+  FocusEventHandler,
+  isValidElement,
+  KeyboardEventHandler,
+  MouseEvent,
+  MouseEventHandler,
+  ReactNode,
+  useEffect,
+  useRef,
+  useState,
+} from "react";
+import {
+  ArrowDown,
+  ArrowUp,
+  End,
+  Home,
+  PageDown,
+  PageUp,
+  Enter,
+  Space,
+} from "../common-hooks";
+import { useEventCallback } from "../utils";
+
+interface UseListProps {
+  children: ReactNode;
+  displayedItemCount: number;
+  // ListNextControlProps
+  onBlur?: FocusEventHandler;
+  onFocus?: FocusEventHandler;
+  onKeyDown?: KeyboardEventHandler;
+  onMouseDown?: MouseEventHandler;
+}
+
+const getSelected = (children: ReactNode): number[] =>
+  Children.toArray(children).reduce(
+    (selectedItems: number[], child: ReactNode, index) => {
+      // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
+      if (isValidElement(child) && child.props?.selected) {
+        selectedItems.push(index);
+      }
+      return selectedItems;
+    },
+    []
+  );
+
+export const useList = ({
+  children,
+  displayedItemCount,
+  onFocus,
+  onKeyDown,
+  onBlur,
+  onMouseDown,
+}: UseListProps) => {
+  const listRef = useRef<HTMLUListElement | null>(null);
+  let list = listRef.current;
+
+  const [activeDescendant, setActiveDescendant] = useState<string>("");
+  const [focusedIndex, setFocusedIndex] = useState<number | null>(null);
+  const [selectedIndexes, setSelectedIndexes] = useState<number[]>(
+    getSelected(children)
+  );
+  const [allOptions, setAllOptions] = useState<Element[]>([]);
+  const [activeOptions, setActiveOptions] = useState<Element[]>([]);
```

Comment:
```suggestion
  const activeOptions = allOptions.filter((option) => option.getAttribute("aria-disabled") !== "true");
```

Reply:
adding this code block here so it's easier for future reference

currently we've been doing `setActiveOptions` in the useEffect: 
```
setActiveOptions(
        Array.from(list.children)
          .filter((child) => child.getAttribute("role") === "option")
          .filter((option) => option.getAttribute("aria-disabled") !== "true")
);
```
`useEffect` will be refactored, will review the recommended approach when we work on the feedback on Monday (fyi @libertymayc)

Reply:
commented this one state is derived from the other, so I dont think it needs to be in state too